### PR TITLE
vine: handling temp files when peer transfers disabled

### PR DESCRIPTION
**Pull request by colinthomas-z80**
## Proposed Changes

If peer transfers are disabled change temp file behavior so they are returned on creation like normal files. Taskvine will still manage naming and cleanup after the workflow completes.

## Merge Checklist

The following items must be completed before PRs can be merge.
Check these off to verify you have completed all steps.

- [ ] `make test`       Run local tests prior to pushing.
- [ ] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [ ] `make lint`       Run lint on source code prior to pushing.
- [ ] Manual Update     Update the manual to reflect user-visible changes.
- [ ] Type Labels       Select a github label for the type: bugfix, enhancement, etc.
- [ ] Product Labels    Select a github label for the product: TaskVine, Makeflow, etc.
- [ ] PR RTM            Mark your PR as ready to merge.

### CODE REVIEW

File: taskvine/src/manager/vine_file.c [hunk 1]
```diff
@@ -245,6 +245,14 @@ struct vine_file *vine_file_temp()
 	return vine_file_create("temp", 0, 0, 0, VINE_TEMP, 0, cache, 0);
 }
 
+struct vine_file *vine_file_pseudo_temp()
+{
+	// temp files are always cached at workers until explicitely removed.
+	vine_cache_level_t cache = VINE_CACHE_LEVEL_WORKFLOW;
+
+	return vine_file_create("temp", 0, 0, 0, VINE_FILE, 0, cache, 0);
```
Comment (btovar) [inline taskvine/src/manager/vine_file.c:253]: I'm not quite sure this one will work. Wouldn't all the temp files get the same name locally at the manager? Also I suggest changing the name to `vine_file_temp_no_peers` or something like that.

**Review by btovar — CHANGES_REQUESTED** [review]

**Comment by colinthomas-z80** [comment]
I am thinking about how to add a mechanism to clean up these local temp files when the workflow completes, to make them similar in function to real temp files

**Comment by dthain** [comment]
https://cctools.readthedocs.io/en/latest/api/html/taskvine_8h.html#a2d4d0d4621e65083488ccef6464f334ca898d527e82997c21744baf54409cce36

**Comment by dthain** [comment]
This would seem to give all temp files from the same user the same name.
```
char *name = string_format("temp-local-%u", uid);
```
Am I missing something?

**Comment by colinthomas-z80** [comment]
You are right. I was mistaken about uid vs. uuid behavior

**Review by dthain — APPROVED** [review]

**Pull request merged**